osu.Game.Tournament/IO/TournamentStorage.cs

```diff
@@ -29,9 +34,22 @@ public TournamentStorage(Storage storage)
             else
                 Migrate(UnderlyingStorage.GetStorageForDirectory(default_tournament));
 
+            CurrentTournament = new Bindable<string>(storageConfig.Get<string>(StorageConfig.CurrentTournament));
```

Comment:
This pattern is super confusing to me and risky (will be wrong if a restart is ever not needed). Maybe just do

```suggestion
            CurrentTournament = storageConfig.GetBindable<string>(StorageConfig.CurrentTournament);
```

Pretty sure that'd spare you the `Set` call in `updateTournament`, too.